spec/components/rails_bootstrap/badge_component_spec.rb

```diff
@@ -0,0 +1,52 @@
+require "rails_helper"
+
+RSpec.describe RailsBootstrap::BadgeComponent, type: :component do 
+  context "variation attribute tests" do
+    it "renders default variation" do 
+      subject = render_inline(RailsBootstrap::BadgeComponent.new()) { "This is a default badge" }
+
+      expect(subject.css("span[class='badge badge-primary']").to_html).not_to be_empty
+    end
+
+    it "renders secondary style" do 
+      subject = render_inline(RailsBootstrap::BadgeComponent.new(variation: "secondary")) { "This is the secondary style badge" }
+
+      expect(subject.css("span[class='badge badge-secondary']").to_html).not_to be_empty
+    end
+
+    it "renders info style" do 
+      subject = render_inline(RailsBootstrap::BadgeComponent.new(variation: "info")) { "This is the info style badge" }
+
+      expect(subject.css("span[class='badge badge-info']").to_html).not_to be_empty
+    end
+
+    it "renders the content" do 
+      subject = render_inline(RailsBootstrap::BadgeComponent.new(variation: "success")) { "This is the success style badge" }
+
+      expect(subject.css("span[class='badge badge-success']").to_html).to include("This is the success style badge")
+    end
+  end 
+
+  context "pill attribute tests" do
```

Comment:
Here is something you should know: 

```
The purpose of “describe” is to wrap a set of tests against one functionality while “context” is to wrap a set of tests against one functionality under the same state. Here’s an example

Source -- https://lmws.net/describe-vs-context-in-rspec
```
Thus context is used when you are dealing with multiple test cases in the same state. Here, you don't need a context block. Just make the test case name more descriptive